src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -2,59 +2,200 @@ import AbstractMapFactory from "./abstract-provider";
 import mapboxgl from "mapbox-gl";
 import "mapbox-gl/dist/mapbox-gl.css";
 
+import VectorTileClient from "../../client/vector-tile-client";
+
 mapboxgl.accessToken = MAP_PROVIDER_TOKEN;
 
-const mapboxGLMethods = {
-  createMap: function(container, options) {
-    options.container = container;
-    this._map = new mapboxgl.Map(options);
+export default (container, options) => {
+  options.map.container = container;
+  options.map.style = {
+    version: 8,
+    sources: {},
+    layers: [],
+    sprite: `${window.location.protocol}//${
+      window.location.host
+    }/static/css/images/markers/spritesheet`,
+  };
 
-    return this;
-  },
+  const map = new mapboxgl.Map(options.map);
+  map.addControl(
+    new mapboxgl.NavigationControl(options.control),
+    options.control.position,
+  );
 
-  on: function(event, callback, context) {
-    callback = callback.bind(context);
-    this._map.on(event, callback);
-  },
+  return AbstractMapFactory({
+    on: (event, callback, context) => {
+      callback = callback.bind(context);
+      map.on(event, callback);
+    },
 
-  getMap: function() {
-    return this._map;
-  },
+    getMap: () => {
+      return map;
+    },
 
-  addNavControl: function({ options, vendorOptions }) {
-    this._navControl = new mapboxgl.NavigationControl(vendorOptions);
-    this._map.addControl(this._navControl, options.position);
+    hasLayer: layerId => {
+      return !!map.getLayer(layerId);
+    },
 
-    return this._navControl;
-  },
+    addMapboxStyle: styleUrl => {
+      map.setStyle(styleUrl);
+    },
 
-  hasLayer: function(layerId) {
-    return !!this._map.getLayer(layerId);
-  },
+    setMaxZoom: zoom => {
+      map.setMaxZoom(zoom);
+    },
 
-  addMapboxStyle: function(styleUrl) {
-    this._map.setStyle(styleUrl);
-  },
+    getZoom: () => {
+      return map.getZoom();
+    },
 
-  setMaxZoom: function(zoom) {
-    this._map.setMaxZoom(zoom);
-  },
+    getBBoxString: () => {
+      return map.getBounds().toString();
+    },
 
-  getZoom: function() {
-    return this._map.getZoom();
-  },
+    getCenter: () => {
+      return map.getCenter();
+    },
 
-  getBBoxString: function() {
-    return this._map.getBounds().toString();
-  },
+    invalidateSize: () => {
+      map.resize();
+    },
 
-  getCenter: function() {
-    return this._map.getCenter();
-  },
+    createRasterTileLayer: options => {
+      return {
+        id: options.id,
+        type: "raster",
+        source: {
+          type: "raster",
+          tiles: [options.url],
+        },
+      };
+    },
 
-  invalidateSize: function() {
-    this._map.resize();
-  },
-};
+    createVectorTileLayer: async options => {
```

Comment:
@modulitos -- does the usage of `async / await` here (and in the new `vector-tile-client`) look correct?

Reply:
Looks good :+1: